### ISSUE TITLE
Upgrade to golang implementation of registry

### ISSUE THREAD

**Issue by pbcole** [mbabineau/cloudformation-docker-registry#1]
On the docker registry page: https://github.com/docker/docker-registry
WARNING
Notice: The classical python "Docker Registry" is deprecated, in favor of a new golang implementation. This here is kept for historical purpose, and will not receive any significant work/love any more. You should head to the landing page of the new registry or the "Distribution" github project instead.

Can the version of the Docker registry be upgraded to the new version?

**Comment by mbabineau** [comment]
Yup, this would be great. I'm swamped right now, but I'd be delighted to review any pull requests!

Any takers? :)

**Comment by mikepii** [comment]
I gave it a shot, but ran into two challenges. The first is that [there isn't a good health check endpoint that's compatible with ELB](https://github.com/docker/distribution/issues/629). The other problem I'm running into is that the docker client on my local machine (version 1.7.1) thinks it's a v1 endpoint and keeps hitting `/v1/users` when trying to authenticate. I found where it [detects the endpoint version](https://github.com/docker/docker/blob/v1.7.1/registry/endpoint.go#L138), but haven't been able to get debug logging working on my system.

I'm setting down this project for now. Here's the work-in-progress, with no ELB health check. It's based on the original template.

https://github.com/RUNDSP/cloudformation-docker-registry/blob/registry2/docker-registry2.json

**Comment by mumoshu** [comment]
Thanks for sharing this 👍 
FYI, I've forked this repo and on top of mikepii's work, managed to support v2.
https://github.com/mumoshu/cloudformation-docker-registry
